core/internal/cltest/cltest.go

```diff
@@ -332,6 +333,19 @@ func NewApplicationWithConfig(t testing.TB, cfg *configtest.TestGeneralConfig, f
 		lggr = logger.TestLogger(t)
 	}
 
+	var auditLogger audit.AuditLogger
+	for _, dep := range flagsAndDeps {
+		audLgger, is := dep.(audit.AuditLogger)
+		if is {
+			auditLogger = audLgger
+			break
+		}
+	}
+
+	if auditLogger == nil {
+		auditLogger = &audit.AuditLoggerService{}
```

Comment:
You might consider using the null pattern instead here.

Reply:
This will mean checking nil at every audit callsite. Is this preferred?

Reply:
I think that Sam meant a `NullAuditLogger` type, equivalent to a `NoopLogger`. If you use a Null/Noop func as mentioned elsewhere, then you can fiddle with the implementation without having to refactor all of these callers.